ddtrace/tracer/spancontext.go

```diff
@@ -236,6 +236,12 @@ func (t *trace) finishedOne(s *span) {
 	if len(t.spans) != t.finished {
 		return
 	}
+
+	// Adjust the starting trace size to avoid allocations.
```

Comment:
The general convention in the repository (and in Go projects in general, to the best of my understanding) is to put comments addressing contents of `if` statements inside the `if` statement, starting with a lower-case letter. 

I'm also a bit unsure if this comment cuts it. Are we avoiding allocations? Seems to me like we are causing them. What we are avoiding is repeated calls to `runtime.growslice`. Maybe the comment should say:

```go
// adjust the starting trace size to avoid repeated calls to runtime.growslice
```